test/marketplace/Abstracts.sol

```diff
@@ -2,17 +2,18 @@
 
 pragma solidity 0.8.4;
 
-abstract contract Erc20 {
-	function approve(address, uint256) virtual external returns (bool);
-	function transfer(address, uint256) virtual external returns (bool);
-	function balanceOf(address) virtual external returns (uint256);
-	function transferFrom(address, address, uint256) virtual public returns (bool);
+interface Erc20 {
+	function approve(address, uint256) external returns (bool);
+	function transfer(address, uint256) external returns (bool);
+	function balanceOf(address) external view returns (uint256);
+	function transferFrom(address, address, uint256) external returns (bool);
+	function decimals() external returns (uint8);
```

Comment:
if we keep interface then this should be view as well

Reply:
👍

Reply:
#178